pkg/stream/xprotocol/connpool_multiplex.go

```diff
@@ -59,24 +69,30 @@ func (p *poolMultiplex) init(client *activeClientMultiplex, sub types.ProtocolNa
 		client, _ := p.newActiveClient(context.Background(), sub)
 		if client != nil {
 			client.state = Connected
-			p.activeClients.Store(sub, client)
+			client.indexInPool = index
+			p.activeClients[index].Store(sub, client)
 		} else {
-			p.activeClients.Delete(sub)
+			p.activeClients[index].Delete(sub)
 		}
 	}, nil)
 }
 
 // CheckAndInit init the connection pool
 func (p *poolMultiplex) CheckAndInit(ctx context.Context) bool {
 	var client *activeClientMultiplex
+	clientIdx := atomic.AddInt64(&p.currentCheckAndInitIdx, 1) % int64(len(p.activeClients))
```

Comment:
a request maybe call the func more than once

Reply:
@taoyuanyuan , this is ok, the caller only return the pool when CheckAndInit returns true, and under this situation, the client slot must be inited and the index is saved to the ctx

Reply:
currently, we'll set the client index to context when CheckAndInit returns true, and use this index to find the client in the pool should be valid